Step01/app/src/main/java/com/kwoolytech/step01/MainActivity.java

```diff
@@ -16,12 +17,15 @@ protected void onCreate(Bundle savedInstanceState) {
         super.onCreate(savedInstanceState);
         setContentView(R.layout.activity_main);
 
-        ArrayAdapter<CharSequence> adapter;
-        adapter = ArrayAdapter.createFromResource(this, R.array.my_company, android.R.layout.simple_list_item_1);
+        ArrayList<String> list = new ArrayList<String>();
+        list.add("Dasan Networks");
+        list.add("SK Telecom");
+        list.add("GE Appliances");
+
+        KwoolyCompanyAdapter adapter = new KwoolyCompanyAdapter(this, R.layout.listviewitem_company,list);
 
         listView = (ListView)findViewById(R.id.listView);
         listView.setAdapter(adapter);
-
         listView.setOnItemClickListener(new AdapterView.OnItemClickListener() {
             @Override
             public void onItemClick(AdapterView<?> parent, View view, int position, long id) {
```

Comment:
Hey @HQHAN, I have a problem on this. After I implemented the ```KwoolyCompanyAdapter```, this listener logic started to not work. So, as you can see, I put listeners for every view objects in the ```KwoolyCompanyAdapter```. Definitely this is not desirable because, every time we want to fix what is going to happen when "clicked", we need to fix the ```KwoolyCompanyAdapter```. Trying to find a cause now.